src/types.js

```diff
@@ -126,6 +126,13 @@ export type Account = {|
    * See the `pushToken` property in `SessionState`, and docs linked there.
    */
   ackedPushToken: string | null,
+
+  /**
+   * When the user last dismissed the server-not-set-up-for-notifs notice.
+   *
+   * `null` if never or if the server reports it's set up for notifications.
```

Comment:
Can we just make this "`null` if never"? That seems simpler to reason about. Then the interaction with whether the server reports it's set up for notifications is for the UI to handle, where it's deciding whether to show a banner.

(And I think the code there in this version does handle that just as one would want.)

Reply:
The other part is aimed at potential issues that could happen when the `realm_push_notifications_enabled` value changes back and forth in the two-week period after a user dismisses the notice.

Here's an example sequence of events, I guess:

1. Tell the admin to fix it
2. Dismiss the banner
3. Admin fixes it
4. A week goes by
5. Admin reverts the fix
6. (Do you want to see the banner right away, or wait another week?)

Reply:
It's very possible this is a small enough edge case that we don't need to worry about it.

Reply:
Hmm, I see.

Yeah, I think ideally in that scenario we'll promptly start showing the banner again -- you may have gotten notifications, so you think it's fixed, and want to know it's broken again. But also it is probably pretty uncommon that it gets fixed and promptly broken again.

The semantics that points to is still a bit different from this jsdoc, though: it'd be that this is when the user last dismissed the notice, since the last time it became applicable, and `null` if they haven't since then.

In particular, in your example at step 5, this value will be `null` even though the user had dismissed the notice (more recently than never), and even though the server doesn't report it's set up for notifications.